### PR TITLE
Tracks: Add Remote Data Blocks in the list of allowed sources

### PR DESCRIPTION
**Pull request by mehmoodak**
## Proposed Changes

Update `@automattic/calypso-analytics` package to add "Remote Data Blocks" in the list of allowed sources.

## Why are these changes being made?

- We are planning to use `@automattic/calypso-analytics` in the "Remote Data Blocks" plugin and, as a result, need to allow it.

## Testing Instructions

* Checkout this branch.
* Link `@automattic/calypso-analytics` package.
* Follow instructions of [this Remote Data Blocks PR](https://github.com/Automattic/remote-data-blocks/pull/185) and verify that events are getting tracked without any error.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/calypso-analytics/src/tracks.ts [hunk 1]
```diff
@@ -215,13 +215,13 @@ export function recordTracksEvent( eventName: string, eventProperties?: any ) {
 		if (
 			! /^calypso(?:_[a-z0-9]+){2,}$/.test( eventName ) &&
 			! /^jetpack(?:_[a-z0-9]+){2,}$/.test( eventName ) &&
+			! /^remotedatablocks(?:_[a-z0-9]+){2,}$/.test( eventName ) &&
 			! /^wpcom_dsp_widget(?:_[a-z0-9]+){2,}$/.test( eventName ) &&
```
Comment (tyxla) [inline packages/calypso-analytics/src/tracks.ts]: Not a blocker for this PR, but a nice potential follow-up: would be nice if we avoid this regex repetition here by having an array of allowed source prefixes declared and checking against that.

File: packages/calypso-analytics/src/tracks.ts [hunk 2]
```diff
@@ -264,12 +264,11 @@ export function recordTracksEvent( eventName: string, eventProperties?: any ) {
 	if (
 		! eventName.startsWith( 'calypso_' ) &&
 		! eventName.startsWith( 'jetpack_' ) &&
+		! eventName.startsWith( 'remotedatablocks_' ) &&
 		! eventName.startsWith( 'wpcom_dsp_widget_' ) &&
```
Comment (tyxla) [inline packages/calypso-analytics/src/tracks.ts]: Same as above.

File: packages/calypso-analytics/src/tracks.ts [hunk 3]
```diff
@@ -264,12 +264,11 @@ export function recordTracksEvent( eventName: string, eventProperties?: any ) {
...
 		! EVENT_NAME_EXCEPTIONS.includes( eventName )
 	) {
-		debug(
-			'- Event name must be prefixed by "calypso_", "jetpack_", or added to `EVENT_NAME_EXCEPTIONS`'
-		);
+		debug( '- Event name must be prefixed by source or added to `EVENT_NAME_EXCEPTIONS`' );
```
Comment (tyxla) [inline packages/calypso-analytics/src/tracks.ts]:
```suggestion
		debug( '- Event name must be prefixed by a known source or added to `EVENT_NAME_EXCEPTIONS`' );
```

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Make sure to publish the new version:
https://github.com/Automattic/wp-calypso/blob/trunk/docs/monorepo.md#publishing

File: packages/calypso-analytics/src/tracks.ts (hunk 1 above)
Comment (mehmoodak) [reply packages/calypso-analytics/src/tracks.ts]: Fixed in [786e273](https://github.com/Automattic/wp-calypso/pull/96007/commits/786e2736bd2d662a8b17a6fcbefce2ddfeb8f7b7). Please have a look.

File: packages/calypso-analytics/src/tracks.ts (hunk 2 above)
Comment (mehmoodak) [reply packages/calypso-analytics/src/tracks.ts]: Fixed in above commit.

File: packages/calypso-analytics/src/tracks.ts (hunk 3 above)
Comment (mehmoodak) [reply packages/calypso-analytics/src/tracks.ts]: Fixed in above commit.

**Review by maxschmeling — APPROVED** [review]

**Comment by mehmoodak** [comment]
> Make sure to publish the new version

Deployed 🚢

**Pull request merged**